core/src/test/scala/unit/kafka/coordinator/group/GroupMetadataManagerTest.scala

```diff
@@ -610,6 +610,51 @@ class GroupMetadataManagerTest {
   }
 
   @Test
+  def testLoadGroupAndOffsetsFromDifferentSegments(): Unit = {
```

Comment:
Thanks for catching this. Just one clarification. It seems the bug is not actually dependent on reading from multiple segments. Multiple reads within the same segment could also cause the issue (which I'm guessing is this line: https://github.com/apache/kafka/pull/3538/files#diff-47448559499d691a7c286fa29d5b202dR600).